view-creator-framework/src/main/java/org/hypertrace/core/viewcreator/pinot/PinotUtils.java

```diff
@@ -165,6 +166,12 @@ public static Schema convertAvroSchemaToPinotSchema(org.apache.avro.Schema avroS
           if (defaultVal == JsonProperties.NULL_VALUE) {
             defaultVal = null;
           }
+          if (!AvroUtils.isSingleValueField(field)
```

Comment:
Do you this to run for MAP types as well? If not, it might be better to move this into the else block on L163. The default value is set to `""` for MAP type explicitly so I'd assume something might be depending on it.

Reply:
So I'm not familiar with the story behind the map type (because it really feels from this file like it's a hack we placed on top of pinot), but maps won't hit this for a couple different reasons - first, pinot's util, `AvroUtils.isSingleValueField` does not treat maps as multi valued - only arrays (see https://github.com/apache/incubator-pinot/blob/e892cb2942f1f41ac7056d977eeca0e5b22897d0/pinot-plugins/pinot-input-format/pinot-avro-base/src/main/java/org/apache/pinot/plugin/inputformat/avro/AvroUtils.java#L229-L236). Second, the default value of a map, based on the above logic, will always be a string, based on the logic on lines 159-162 above - so none of the conditions are reachable. 

On top of all that, based on my current understanding of the code around defaults, if the map code fell into here, it would actually be _more_ correct (because a map is being split into two multivalued string cols, each of which would make sense to default to null). Setting an empty string I suspect has the same effect due to pinot's defaults for a string data type, but haven't confirmed because in practice we never use a default value for any of our map cols. In other words, we're generating:
```json
    {
      "name": "tags__KEYS",
      "dataType": "STRING",
      "singleValueField": false,
      "defaultNullValue": ""
    }
```
instead of 
```json
    {
      "name": "tags__KEYS",
      "dataType": "STRING",
      "singleValueField": false
    }
```

The reason I'd rather keep the logic after the map block you mentioned is the code feels separated into first initializing a default value, then normalizing the default values for pinot - and this feels more like the latter.

So all that said, happy to either add a comment or rearrange for clarity - @buchi-busireddy what do you think?

Reply:
Your reason makes sense to me so I think we should have comments at the least on why it's added later. Otherwise, I'm fine. 
We should probably fix the map logic at some point and make it simpler.

Reply:
left the empty string with a todo, but otherwise rearranged the logic so the default value overrides exist in sequence.